src/lib/protocols/softether.c

```diff
@@ -104,6 +104,10 @@ static size_t dissect_softether_type(enum softether_value_type t,
       }
       if (t == VALUE_DATA)
       {
+        if (siz == 0)
```

Comment:
Wouldn't it be better to move that condition after the assignment?

Reply:
Something like
```
if (t == VALUE_DATA)
{
  siz--;
}
if(size == (u_int32_t)-1)
  return 0;
```
?

Reply:
I was thinking about:
```C
u_int32_t siz = ntohl(get_u_int32_t(payload, 0));
if (siz == 0) {
  return 0;
}
```

Reply:
While 0 as value for `siz` is clearly wrong for VALUE_DATA, is it invalid also for the other types?

Reply:
AFAIS the Softether protocol does not allow empty array types.

Reply:
ok, thanks for the explanation